src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,258 @@
+    }
+
+    /**
+    @notice function so the bridge can receive ether. Used for subsidy.
+    */
+    receive() external payable {}
+
+    /**
+    @notice Stake and unstake Curve LP tokens through Convex Finance Booster anytime.
+    @notice Convert rate between Curve LP token and corresponding Convex LP token is 1:1
+    @notice Stake == Deposit, Unstake == Withdraw
+    @param _inputAssetA Curve LP token (staking), virtual asset (unstaking)
+    @param _outputAssetA Virtual asset (staking), Curve LP token (unstaking)
+    @param _totalInputValue Total number of Curve LP tokens to deposit / withdraw
+    @param _auxData Data to claim staking rewards
+    @param outputValueA Number of Curve LP tokens staked / unstaked
+    @param _rollupBeneficiary Address of the contract that receives subsidy
+    */
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata _inputAssetB,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata _outputAssetB,
+        uint256 _totalInputValue,
+        uint256,
+        uint64 _auxData,
+        address _rollupBeneficiary
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_totalInputValue == 0) {
+            revert ErrorLib.InvalidInputAmount();
+        }
+
+        _loadPools();
+
+        PoolInfo memory selectedPool;
+
+        if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.ERC20 &&
+            _outputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+        ) {
+            // deposit
+            if (!pools[_inputAssetA.erc20Address].exists) {
+                revert ErrorLib.InvalidInputA();
+            }
+
+            selectedPool = pools[_inputAssetA.erc20Address];
+
+            _setTokens(selectedPool);
+
+            // approvals
+            curveLpToken.approve(address(BOOSTER), _totalInputValue);
+
+            uint256 startCurveRewards = curveRewards.balanceOf(address(this));
+
+            BOOSTER.deposit(selectedPool.poolId, _totalInputValue, true);
+
+            uint256 endCurveRewards = curveRewards.balanceOf(address(this));
+
+            outputValueA = (endCurveRewards - startCurveRewards);
+
+            interactions[_outputAssetA.id] = Interaction(_totalInputValue, selectedPool.convexToken, true);
+        } else if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL &&
+            _outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20
+        ) {
+            // withdraw
+            if (!interactions[_inputAssetA.id].exists) {
+                revert UnknownVirtualAsset();
+            }
+            // withdraw amount needs to be equal to the staked value of the interaction
+            if (interactions[_inputAssetA.id].valueStaked != _totalInputValue) {
```

Comment:
This is not feasible because you can't ensure that all the users which were present in the interaction would exit at the same time. I realized that unfortunately the virtual assets were not a good idea here because they would currently break aggregation --> virtual asset id is set to `interactionNonce` which makes it impossible to aggregate users holding virtualAssets corresponding to the same pool when interactionNonce is different. Doing the change should be fairly straighforward though. In the `_loadPools(...)` function deploy a new ERC20 for each pool and in convert function be minting and burning the token. You want to deploy mintable ERC20 where the bridge will have the rights to mint and burn. Check out Openzeppelin's ERC20 token implementations.